TestDataAccess/JSONReader.cs

```diff
@@ -92,12 +91,23 @@ private static JObject ConvertJSONFileToJObject(JSONFile jsonFile)
         /// <param name="testDataKey">key property for the testdata</param>
         /// <param name="index">Index of the testcase(s)</param>
         /// <returns></returns>
-        private JObject GetKeyAndIndexValueFromJObject(JSONFile jsonFile,
+        private JObject GetKeyAndIndexValueFromJObject(
             string testDataKey, int index)
         {
-            return (JObject)ConvertJSONFileToJObject(jsonFile)
-                .GetValue(testDataKey)
-                .ElementAt(index);
+            var jObject = this.ConvertJSONFileToJObject();
+            var value = jObject.GetValue(testDataKey);
+            var element = value.ElementAt(index);
+
+            return (JObject)element;
```

Comment:
This was causing a cast error (sometimes) when the caller was expecting a _single_ value from the JSON. This seems to work if its looking for an object or an array. 

Adding GetJsonPropertyValue() helps resolve, but it indicates a bigger problem... that this function isn't being used correctly.